react/components/EditorContainer/StoreEditor/Styles/StyleEditor/typography/FileFontEditor.tsx

```diff
@@ -0,0 +1,195 @@
+        'Uploaded files must be under 2MB and have one of the extensions: ' +
+          FONT_FILE_EXTENSIONS.join(' ')
+      )
+    }
+
+    if (acceptedFiles.length > 0) {
+      dispatchFiles({
+        files: acceptedFiles.map(fontFile => ({ fontFile })),
+        type: 'append',
+      })
+    }
+  }
+
+  const isUpload = (file: FontFileInput): file is FontFileUpload =>
+    (file as FontFileUpload).fontFile != null
+
+  const getFileName = (file: FontFileInput): string =>
+    isUpload(file) ? file.fontFile.name : file.filename
+
+  return (
+    <div className="mv6 w-100">
+      <Input
+        size="small"
+        label={
+          <FormattedMessage
+            id="admin/pages.editor.styles.edit.font-family.title"
+            defaultMessage="Font Family"
+          />
+        }
+        value={family}
+        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+          setFamily(e.target.value)
+        }
+      />
+      {files.map((file, index) => (
+        <FontFileItem
+          fileName={getFileName(file)}
+          onRemove={() => removeFile(index)}
+          onStyleUpdate={(style: FontFlavour) => updateFile(index, style)}
+          style={getFontFlavour(files[index])}
+        />
+      ))}
+      <FontFileUploadComponent onDrop={onDrop} />
+    </div>
+  )
+}
+
+interface FontFileItemProps {
+  fileName: string
+  onRemove: () => void
+  onStyleUpdate: (style: FontFlavour) => void
+  style?: FontFlavour
+}
+
+const FontFileItem: React.FunctionComponent<FontFileItemProps> = ({
+  fileName,
+  onRemove,
+  onStyleUpdate,
+  style,
+}) => {
+  const actionMenuProps = {
+    buttonProps: {
+      icon: <IconOptionsDots color="currentColor" />,
+      variation: 'tertiary',
+    },
+    options: [
+      {
+        label: (
+          <FormattedMessage
+            id="admin/pages.admin.redirects.form.button.remove"
+            defaultMessage="Remove"
+          />
+        ),
+        onClick: onRemove,
+      },
+    ],
+  }
+
+  const dropdownProps = {
+    onChange: (_: React.ChangeEvent, value: string) =>
+      onStyleUpdate(stringToFlavour(value)),
+    options: STYLE_FLAVOUR_OPTIONS,
+    value: flavourToString(style),
+    variation: 'inline',
+  }
+
+  return (
+    <div className="mv5">
+      <div className="flex items-center">
+        <div className="mv3">{fileName}</div>
+        <div className="flex-grow-1" />
+        <ActionMenu {...actionMenuProps} />
+      </div>
+      <div className="flex items-center">
+        <FormattedMessage
+          id="admin/pages.editor.styles.edit.custom-font.style"
+          defaultMessage="Style:"
+        />
+        <Dropdown {...dropdownProps} />
+      </div>
+    </div>
+  )
+}
+
+interface FontFileUploadProps {
+  onDrop: DropzoneOptions['onDrop']
+}
+
+const FontFileUploadComponent: React.FunctionComponent<FontFileUploadProps> = ({
+  onDrop,
+}) => {
+  const { getRootProps, getInputProps } = useDropzone({
+    accept: FONT_FILE_EXTENSIONS,
```

Comment:
Is passing the extension correct here? I thought you had to pass a MIME type here. For [fonts](https://www.iana.org/assignments/media-types/media-types.xhtml#font), it would be:

Name | Template
-- | --
collection | font/collection
otf | font/otf
sfnt | font/sfnt
ttf | font/ttf
woff | font/woff
woff2 | font/woff2

Reply:
It isn't working with MIME types :(